src/v/kafka/server/snc_quota_manager.cc

```diff
@@ -126,11 +140,28 @@ quota_t node_to_shard_quota(const std::optional<quota_t> node_quota) {
     }
 }
 
+auto update_node_bucket(
+  ssx::sharded_ptr<kafka::snc_quota_manager::bucket_t>& b, auto const& cfg) {
```

Comment:
Does this method need to be a template? Looks like `cfg` is always the same type.

Reply:
Good catch, I refactored from a lambda with an auto param (which is a habit of mine).

Reply:
Done